CGAT/scripts/bam2libtype.py

```diff
@@ -0,0 +1,163 @@
+"""bam2libtype.py - determine the library type of a bam file
+============================================================
+
+Author: Adam Cribbs
+
+Purpose
+-------
+
+This tool determines the library type of a BAM file. The naming
+convention used is from the salmon documentation:
+http://salmon.readthedocs.io/en/latest/library_type.html.
+
+BAM files need to have a corresponding index file i.e. example.bam
+and example.bam.bai
+
+
+Usage
+-----
+
+    cat example.bam | cgat bam2libtype > out.tsv
+
+options
+-------
+
+There are no options for this script, just pass the script a bam file
+as the stdin and an outfile as the stdout.
+
+
+Type::
+
+   python bam2bed.py --help
+
+for command line help.
+
+Command line options
+--------------------
+"""
+
+import sys
+import pysam
+import CGAT.Experiment as E
+
+
+def main(argv=None):
+    """script main.
+
+    parses command line options in sys.argv, unless *argv* is given.
+    """
+
+    if not argv:
+        argv = sys.argv
+
+    # setup command line parser
+    parser = E.OptionParser(
+        version="%prog version: $Id$", usage=globals()["__doc__"])
+
+    (options, args) = E.Start(parser, argv=argv)
+
+    iterations = "1000000"
+    samfile = pysam.AlignmentFile(options.stdin, "rb")
+    outfile = options.stdout
+
+    n = 0
+
+    # initialise counts for each library type
+    MSR = 0
+    MSF = 0
+    ISF = 0
+    ISR = 0
+    OSF = 0
+    OSR = 0
+    SR = 0
+    SF = 0
+
+    reads_processed = set()
+
+    for read in samfile:
+
+        n += 1
+        if n <= int(iterations):
+
+            # to handle paired end reads:
+            if read.is_paired and read.is_proper_pair:
+                if read.qname in reads_processed:
```

Comment:
* `reads_processed` is initialised as an emtpy set and never modified so this statement will never be true, is it required then?